proposals/0296-increase-tx-size.md

```diff
@@ -0,0 +1,172 @@
+ For example bit 0 may mean "requested_cus", bit 1 may mean "microlamports per
+ cu"
+RecentBlockhash (hash)
+NumAddresses (u8)
+Addresses [[u8; 32]] -- Number matches NumAddresses
+ResourceRequests [u64] -- Array of request values. (section size is popcount
+ ResourceRequestMask * 8). each value is a u64.
+Ixs [(u16, u16, u16, u16)] -- Number matches NumInstructions. Values are
+ (num_accounts, accounts_offset, num_data_bytes, bytes_offset).
+[TRAILING DATA SECTION] -- Length is such that PayloadLength is matched.
+Signatures [[u8; 64]] -- Length of `num_required_signatures` from
+ `LegacyHeader`
+```
+
+This new `v1` transaction format notably does not include address lookup
+tables.
+
+In consideration of what size the new transaction size limit should increase
+to, [Jito's bundle
+
+sizes](https://jitolabs.grafana.net/dashboard/snapshot/ISnwjbxw02UBLrj1xy4n4dFkAPyZ46ll?orgId=0&from=2025-05-30T18:45:00.000Z&to=2025-05-30T21:45:00.000Z&timezone=utc&var-cluster=mainnet&var-region=$__all&viewPanel=panel-40&inspect=panel-40&inspectTab=data)
+that were used to increase the overall transaction size should be considered.
+The following is the rough distribution of the transaction sizes:
+
+- 2048 bytes or lower - 50% of bundles
+- 6144 bytes or lower = 65% of all bundles
+- 9216 bytes or lower = 100% of all bundles
+
+In consideration of the above, the use cases mentioned within the motivation
+section, and the current page size for managing memory within a validator, a
+new transaction size limit of 4096 bytes is proposed. This new limit should
+cover a majority of use cases mentioned within the motivation section, as well
+as enable most of the developers using Jito bundles for larger transactions to
+migrate to the new transaction format. Similarly, since the new limit proposed
+can accomodate the max accounts used in ALTs directly in the transaction,
+developers are able to migrate from `v0` transactions to `v1` transactions.
+
+A number of changes are required to be made to the validator to support the new
+transaction size limit. Consensus would need to be updated to support the
+larger transaction size, as larger transactions included in a block would be
+marked as invalid by the cluster today. The scheduler would also need to be
+modified to  support the larger transaction sizes. While this proposal does not
+introduce a new fee structure around bytes in a transaction, the scheduler
+should prioritize larger transactions differently, requiring developers to pay
+a higher priority fee to land their transaction.
+
+Testing for this new transaction size limit should be done extensively to
+ensure that performance on the cluster is not adversely affected. This testing
+should include the different use cases mentioned within the motivation section,
+as well as the different sizes of transactions that are currently being used by
+developers on the network.
+
+## Alternatives Considered
+
+Alternative designs considered:
+
```

Comment:
I would like to add changes to update the `struct Packet` keeping the same size for 1280 for shreds gossip and turbine and a new struct for Transaction. Currently we use struct Packet and PACKET_DATA_SIZE everywhere which creates a lot of confusion. PACKET_DATA_SIZE should be left to 1232 while only for transaction size should be updated.